som_crm/models/res_users.py

```diff
@@ -23,3 +30,21 @@ def _check_call_center_user(self):
             )
             if count:
                 raise ValidationError(_("The field 'Call center user' must be unique."))
+
+    def _get_assigned_opportunities_count(self):
+        """ Returns the number of opportunities assigned to the user. """
+        self.ensure_one()
+        return self.env['crm.lead'].search_count([
+            ('user_id', '=', self.id),
+            ('type', '=', 'opportunity'),
+        ])
+
+    def _get_available_leads_capacity(self):
+        """ Returns the number of available leads capacity for the user. """
+        self.ensure_one()
+        if self.som_max_leads_capacity == 0:
```

Comment:
We should consider using -1 to represent infinity. This allows us to reserve 0 for users who are present but should not receive opportunities.